source/javascripts/components/StacksAndMachine/MachineTypeSelector.tsx

```diff
@@ -28,134 +19,120 @@ export const promotionTexts = {
   onChange,
-  promotedOptions,
 }: Props) => {
   const machineTypePromotion = useMemo(() => {
     if (!machineTypePromotionMode) {
       return null;
     }
 
-    return promotionTexts[machineTypePromotionMode];
+    return PROMOTION_TEXTS[machineTypePromotionMode];
   }, [machineTypePromotionMode]);
 
-  const toggletip = (icon: ReactNode) => {
-    if (!machineTypePromotion) {
-      return null;
+  // TODO: Dropdown should handle this somehow
+  // const toggletip = (icon: ReactNode) => {
+  //   if (!machineTypePromotion) {
+  //     return null;
+  //   }
+
+  //   return (
+  //     <Toggletip
+  //       button={{
+  //         href: 'https://bitrise.io/demo',
+  //         label: 'Reach out to sales',
+  //       }}
+  //       label={machineTypePromotion.toggleTipText}
+  //     >
+  //       {icon}
+  //     </Toggletip>
+  //   );
+  // };
+
+  const groups = useMemo(() => {
+    if (!machineTypePromotion || !promotedOptions || promotedOptions.length === 0) {
+      return [
+        {
+          isEnabled: true,
+          label: 'Available on your plan',
+          machines: availableOptions,
+        },
+      ];
     }
 
-    return (
-      <Toggletip
-        button={{
-          href: 'https://bitrise.io/demo',
-          label: 'Reach out to sales',
-        }}
-        label={machineTypePromotion.toggleTipText}
-      >
-        {icon}
-      </Toggletip>
-    );
-  };
+    return [
+      {
+        isEnabled: true,
+        label: machineTypePromotion.availableText,
+        machines: availableOptions,
+      },
+      {
+        isEnabled: false,
+        label: machineTypePromotion.promotedText,
+        machines: promotedOptions,
+      },
+    ];
+  }, [availableOptions, machineTypePromotion, promotedOptions]);
 
   return (
     <Dropdown
+      flex="1"
+      required
+      search={false}
+      label="Machine type"
       disabled={isLoading || isDisabled}
+      infoTooltipLabel={machineTypePromotion?.toggleTipText}
       errorText={isInvalid ? 'Invalid machine type' : undefined}
       helperText={machineType.creditPerMinute ? `${machineType.creditPerMinute} credits/min` : undefined}
-      label="Machine type"
-      labelHelp={toggletip}
       onChange={(e) => onChange(e.target.value as string)}
-      required
-      search={false}
       value={machineType.value}
     >
-      {machineTypePromotion && promotedOptions && promotedOptions.length > 0
-        ? [
-            {
-              isEnabled: true,
-              label: machineTypePromotion.availableText,
-              machines: availableOptions,
-            },
-            {
-              isEnabled: false,
-              label: machineTypePromotion.promotedText,
-              machines: promotedOptions,
-            },
-          ].map(({ isEnabled, label, machines }) => {
+      {groups.map(({ label, machines, isEnabled }) => (
+        <DropdownGroup key={label} label={label}>
+          {machines.map((machine) => {
```

Comment:
This is a map within a map again.

Also, this will display a single group when there is no promotion. It was a product decision to not have such.

Reply:
Unfortunately, the Dropdown component can't handle if the `Dropdown*` subcomponents are not inlined directly...